### PR TITLE
Update decomposition of MultiControlPauli to use AND/AND† for num_controls=2

### PR DESCRIPTION
**Pull request by tanujkhattar**
Part of fixing https://github.com/quantumlib/Qualtran/issues/785

The `MultiControlPauli` should stay true to it's description of using `n - 1` clean ancillas for `n` controls and using $\text{And} / \text{And}^{\dagger}$  ladders with a singly controlled pauli at the end.

### CODE REVIEW

**Comment by mpharrigan** [comment]
What was happening before? It was using `MultiAnd` with two control values? Should that be an error or somehow force a replacement with `[TwoBit]And`?

**Comment by tanujkhattar** [comment]
It was using cirq.CCX, cirq.CCZ etc for 2 control case

File: qualtran/bloqs/mcmt/multi_control_multi_target_pauli.py [hunk 1]
```diff
@@ -123,20 +123,25 @@ def decompose_from_registers(
         self, *, context: cirq.DecompositionContext, **quregs: NDArray['cirq.Qid']
     ) -> cirq.OP_TREE:
         controls, target = quregs.get('controls', np.array([])), quregs['target']
-        if len(self.cvs) <= 2:
+        if len(self.cvs) < 2:
```
Comment (mpharrigan) [inline qualtran/bloqs/mcmt/multi_control_multi_target_pauli.py:126]: do you still need `controls.flatten()`? Does this work for len(self.cvs) == 0?

**Review by mpharrigan — COMMENTED** [review]
Can you take the opportunity to flesh out the docstring for this class a bit. In particular: which part of the reference does this correspond to?

**Comment by mpharrigan** [comment]
Does this actually use the fact that the gate is a Pauli? Or does it work for any single qubit gate?

File: qualtran/bloqs/mcmt/multi_control_multi_target_pauli.py (hunk 1 above)
Comment (tanujkhattar) [reply qualtran/bloqs/mcmt/multi_control_multi_target_pauli.py:126]: Yes and Yes. We have tests already for `len(self.cvs) == 0` which are passing after this PR as well.

**Comment by tanujkhattar** [comment]
> Can you take the opportunity to flesh out the docstring for this class a bit. In particular: which part of the reference does this correspond to?

@mpharrigan I've added more details to the docstring. 

> 
Does this actually use the fact that the gate is a Pauli? Or does it work for any single qubit gate?

It works for any single qubit gate. In a follow-up PR we can factor out the logic into a generic `MultiControlBloq` that delegates to a singly controlled Bloq between an $\text{AND} / \text{AND}^\dagger$ ladder

**Review by mpharrigan — APPROVED** [review]

**Pull request merged**